pdr_backend/sim/sim_engine.py

```diff
@@ -147,7 +165,19 @@ def run_one_iter(self, test_i: int, mergedohlcv_df: pl.DataFrame):
         ut = UnixTimeMs(recent_ut - testshift * timeframe.ms)
 
         # predict price direction
-        prob_up: float = self.model.predict_ptrue(X_test)[0]  # in [0.0, 1.0]
+        if self.ppss.sim_ss.use_own_model is not False:
+            prob_up: float = self.model.predict_ptrue(X_test)[0]  # in [0.0, 1.0]
+        else:
+            ut_seconds = ut.to_seconds()
+            if (
+                self.prediction_dataset is not None
+                and ut_seconds in self.prediction_dataset
+            ):
+                # check if the current slot is in the keys
+                prob_up = self.prediction_dataset[ut_seconds]
+            else:
```

Comment:
let's log something here. I'm also not very happy about the imbrication.
Maybe move ut_seconds before, and use `else if`?